test/src/testPubSub.cpp

```diff
@@ -0,0 +1,267 @@
+            }
+            return make_tuple(0, data, end_time);
+        }
+    }
+};
+
+static random_device rand_dev;
+static default_random_engine rand_eng(rand_dev());
+
+static std::vector<uint8_t> get_random_vec(size_t len)
+{
+    static uniform_int_distribution<> rand_dist(0, 0xff);
+    vector<uint8_t> results(len);
+    for (size_t i = 0; i < len; i++) results[i] = rand_dist(rand_eng);
+    return results;
+}
+
+static size_t get_random_length()
+{
+    static uniform_int_distribution<> rand_dist(1, 1400);
+    return rand_dist(rand_eng);
+}
+
+struct TestPubSub : public ::testing::Test {
+        // SetUpTestSuite() is called before all tests in the test suite
+        static void SetUpTestSuite() {}
+
+        // TearDownTestSuite() is called after all tests in the test suite
+        static void TearDownTestSuite() {}
+};
+
+TEST_F(TestPubSub, interprocess) {
+    using namespace std::chrono;
+
+    auto uuri = BuildUUri()
+                      .setAutority(BuildUAuthority().build())
+                      .setEntity(BuildUEntity()
+                              .setMajorVersion(1)
+                              .setId(1)
+                              .build())
+                      .setResource(BuildUResource()
+                              .setID(1)
+                              .build())
+                      .build();
+
+    MessageCapture<true> callback;
+    auto child_pid = fork();
+    if (child_pid == 0) {
+        auto transport = upZenohClient::instance();
+        UStatus listen_status = transport->registerListener(uuri, callback);
+        if (UCode::OK != listen_status.code()) {
+            cerr << "child process listen failed" << endl;
+            exit(-1);
+        }
+        sleep(100000);
+        exit(0);
+    }
+    else {
+        sleep(1);
+        auto transport = upZenohClient::instance();
+
+        auto builder = UAttributesBuilder::publish(uuri, UPriority::UPRIORITY_CS0);
+        UAttributes attributes = builder.build();
+
+        const size_t cnt = 1000;
+        for (size_t i = 0; i < cnt; i++) {
+            auto out_data = get_random_vec(get_random_length());
+            // cout << "test length = " << out_data.size() << endl;
+            UPayload payload((uint8_t*)out_data.data(), out_data.size(), UPayloadType::VALUE);
+            UMessage message(payload, attributes);
+            auto send_time = steady_clock::now();
+            UStatus send_status = transport->send(message);
+            EXPECT_EQ(UCode::OK, send_status.code());
+            auto [ sz, in_data, cap_time ] = callback.get(2000);
+            EXPECT_EQ(1, sz);
+            EXPECT_EQ(out_data.size(), in_data.size());
+            EXPECT_EQ(out_data == in_data, true);
+            // cout << "us = " << duration_cast<microseconds>(cap_time - send_time).count() << endl;
+        }
+        // listen_status = transport->unregisterListener(uuri, callback);
+        // EXPECT_EQ(UCode::OK, listen_status.code());
```

Comment:
It is a reminder of the fact that child process cleanup is not practical, or not worth implementing, and that child process errors cannot practically be directed back to the parent test. This could be fixed but there are bigger fish to fry, I think.